parastell/magnet_coils.py

```diff
@@ -401,164 +386,110 @@ class MagnetCoil(object):
-        self.shape = shape
-        self.shape_str = shape_str
-
-    def _orient_rectangle(
-        self, path_origin, surf_id, t_vec, norm, rot_axis, rot_ang_norm
-    ):
-        """Orients rectangular cross-section in the normal plane such that its
-        thickness direction faces the origin.
-        (Internal function not intended to be called externally)
-
-        Arguments:
-            path_origin (int): index of initial point in filament path.
-            surf_id (int): index of cross-section surface.
-            t_vec (list of float): cross-section thickness vector.
-            norm (list of float): cross-section normal vector.
-            rot_axis (list of float): axis about which to rotate the
-                cross-section.
-            rot_ang_norm (float): angle by which cross-section was rotated to
-                align its normal with the initial point tangent (deg).
-        """
-        # Determine orientation of thickness vector after cross-section was
-        # oriented along filament origin tangent
-
-        # Compute part of thickness vector parallel to rotation axis
-        t_vec_par = normalize(np.inner(t_vec, rot_axis) * rot_axis)
-        # Compute part of thickness vector orthogonal to rotation axis
-        t_vec_perp = normalize(t_vec - t_vec_par)
-
-        # Compute vector othogonal to both rotation axis and orthogonal
-        # part of thickness vector
-        orth = normalize(np.cross(rot_axis, t_vec_perp))
-
-        # Determine part of rotated vector parallel to original
-        rot_par = np.cos(rot_ang_norm)
-        # Determine part of rotated vector orthogonal to original
-        rot_perp = np.sin(rot_ang_norm)
+    def __init__(self, coords, tangents, center_of_mass, width, thickness):
 
-        # Compute orthogonal part of thickness vector after rotation
-        t_vec_perp_rot = rot_par * t_vec_perp + rot_perp * orth
-        # Compute thickness vector after rotation
-        t_vec_rot = normalize(t_vec_perp_rot + t_vec_par)
-
-        # Orient cross-section in its plane such that it faces the global origin
-
-        # Extract initial path point
-        pos = cubit.vertex(path_origin).coordinates()
-
-        # Project position vector onto cross-section
-        pos_proj = normalize(pos - np.inner(pos, norm) * norm)
-
-        # Compute angle by which to rotate cross-section such that it faces the
-        # origin
-        rot_ang_orig = np.arccos(np.inner(pos_proj, t_vec_rot))
-
-        # Re-orient rotated cross-section such that thickness vector faces
-        # origin
-        cubit.cmd(
-            f"rotate Surface {surf_id} angle {np.rad2deg(rot_ang_orig)} about "
-            "origin 0 0 0 direction " + " ".join(str(i) for i in norm)
-        )
+        self.coords = coords
+        self.tangents = tangents
+        self.center_of_mass = center_of_mass
+        self.width = width
+        self.thickness = thickness
 
     def create_magnet(self):
```

Comment:
We need to redo this with everything originally as a numpy array

Reply:
I believe each of those, except `width` and `thickness` of course, are numpy arrays